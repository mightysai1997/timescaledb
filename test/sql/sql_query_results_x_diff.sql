-- This file and its contents are licensed under the Apache License 2.0.
-- Please see the included NOTICE for copyright information and
-- LICENSE-APACHE for a copy of the license.

--make sure diff only has explain output not result output
\! diff ${TEST_OUTPUT_DIR}/results/sql_query_results_optimized.out ${TEST_OUTPUT_DIR}/results/sql_query_results_unoptimized.out
