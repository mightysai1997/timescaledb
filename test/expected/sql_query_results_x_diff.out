--make sure diff only has explain output not result output
\! diff ../results/sql_query_results_optimized.out ../results/sql_query_results_unoptimized.out 
9a10
> SET timescaledb.disable_optimizations= 'on';
86,87c87,88
<                                    QUERY PLAN                                    
< ---------------------------------------------------------------------------------
---
>                                 QUERY PLAN                                 
> ---------------------------------------------------------------------------
91,96c92,98
<          ->  Result
<                ->  Merge Append
<                      Sort Key: (date_trunc('minute'::text, hyper_1."time")) DESC
<                      ->  Index Scan using time_plain on hyper_1
<                      ->  Index Scan using "1-time_plain" on _hyper_1_1_chunk
< (8 rows)
---
>          ->  Sort
>                Sort Key: (date_trunc('minute'::text, hyper_1."time")) DESC
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1
>                            ->  Seq Scan on _hyper_1_1_chunk
> (9 rows)
121,122c123,124
<                                                 QUERY PLAN                                                 
< -----------------------------------------------------------------------------------------------------------
---
>                                                       QUERY PLAN                                                       
> -----------------------------------------------------------------------------------------------------------------------
124,133c126,138
<    ->  GroupAggregate
<          Group Key: (date_trunc('minute'::text, hyper_1."time"))
<          ->  Result
<                ->  Merge Append
<                      Sort Key: (date_trunc('minute'::text, hyper_1."time")) DESC
<                      ->  Index Scan using time_plain on hyper_1
<                            Index Cond: ("time" < 'Wed Dec 31 16:15:00 1969 PST'::timestamp with time zone)
<                      ->  Index Scan using "1-time_plain" on _hyper_1_1_chunk
<                            Index Cond: ("time" < 'Wed Dec 31 16:15:00 1969 PST'::timestamp with time zone)
< (10 rows)
---
>    ->  Sort
>          Sort Key: (date_trunc('minute'::text, hyper_1."time")) DESC
>          ->  HashAggregate
>                Group Key: date_trunc('minute'::text, hyper_1."time")
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1
>                                  Filter: ("time" < 'Wed Dec 31 16:15:00 1969 PST'::timestamp with time zone)
>                            ->  Bitmap Heap Scan on _hyper_1_1_chunk
>                                  Recheck Cond: ("time" < 'Wed Dec 31 16:15:00 1969 PST'::timestamp with time zone)
>                                  ->  Bitmap Index Scan on "1-time_plain"
>                                        Index Cond: ("time" < 'Wed Dec 31 16:15:00 1969 PST'::timestamp with time zone)
> (13 rows)
196,197c201,202
<                                       QUERY PLAN                                       
< ---------------------------------------------------------------------------------------
---
>                                    QUERY PLAN                                    
> ---------------------------------------------------------------------------------
201,206c206,212
<          ->  Result
<                ->  Merge Append
<                      Sort Key: (time_bucket('@ 1 min'::interval, hyper_1."time")) DESC
<                      ->  Index Scan using time_plain on hyper_1
<                      ->  Index Scan using "7-time_plain" on _hyper_1_1_chunk
< (8 rows)
---
>          ->  Sort
>                Sort Key: (time_bucket('@ 1 min'::interval, hyper_1."time")) DESC
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1
>                            ->  Seq Scan on _hyper_1_1_chunk
> (9 rows)
218,219c224,225
<                                                                 QUERY PLAN                                                                 
< -------------------------------------------------------------------------------------------------------------------------------------------
---
>                                                              QUERY PLAN                                                              
> -------------------------------------------------------------------------------------------------------------------------------------
223,228c229,235
<          ->  Result
<                ->  Merge Append
<                      Sort Key: ((time_bucket('@ 1 min'::interval, (hyper_1."time" - '@ 30 secs'::interval)) + '@ 30 secs'::interval)) DESC
<                      ->  Index Scan using time_plain on hyper_1
<                      ->  Index Scan using "7-time_plain" on _hyper_1_1_chunk
< (8 rows)
---
>          ->  Sort
>                Sort Key: ((time_bucket('@ 1 min'::interval, (hyper_1."time" - '@ 30 secs'::interval)) + '@ 30 secs'::interval)) DESC
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1
>                            ->  Seq Scan on _hyper_1_1_chunk
> (9 rows)
240,241c247,248
<                                                    QUERY PLAN                                                    
< -----------------------------------------------------------------------------------------------------------------
---
>                                                 QUERY PLAN                                                 
> -----------------------------------------------------------------------------------------------------------
245,250c252,258
<          ->  Result
<                ->  Merge Append
<                      Sort Key: (time_bucket('@ 1 min'::interval, (hyper_1."time" - '@ 30 secs'::interval))) DESC
<                      ->  Index Scan using time_plain on hyper_1
<                      ->  Index Scan using "7-time_plain" on _hyper_1_1_chunk
< (8 rows)
---
>          ->  Sort
>                Sort Key: (time_bucket('@ 1 min'::interval, (hyper_1."time" - '@ 30 secs'::interval))) DESC
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1
>                            ->  Seq Scan on _hyper_1_1_chunk
> (9 rows)
262,263c270,271
<                                                                 QUERY PLAN                                                                 
< -------------------------------------------------------------------------------------------------------------------------------------------
---
>                                                              QUERY PLAN                                                              
> -------------------------------------------------------------------------------------------------------------------------------------
267,272c275,281
<          ->  Result
<                ->  Merge Append
<                      Sort Key: ((time_bucket('@ 1 min'::interval, (hyper_1."time" - '@ 30 secs'::interval)) + '@ 30 secs'::interval)) DESC
<                      ->  Index Scan using time_plain on hyper_1
<                      ->  Index Scan using "7-time_plain" on _hyper_1_1_chunk
< (8 rows)
---
>          ->  Sort
>                Sort Key: ((time_bucket('@ 1 min'::interval, (hyper_1."time" - '@ 30 secs'::interval)) + '@ 30 secs'::interval)) DESC
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1
>                            ->  Seq Scan on _hyper_1_1_chunk
> (9 rows)
284,285c293,294
<                                         QUERY PLAN                                        
< ------------------------------------------------------------------------------------------
---
>                                      QUERY PLAN                                     
> ------------------------------------------------------------------------------------
289,294c298,304
<          ->  Result
<                ->  Merge Append
<                      Sort Key: (time_bucket('@ 1 min'::interval, hyper_1_tz."time")) DESC
<                      ->  Index Scan using time_plain_tz on hyper_1_tz
<                      ->  Index Scan using "2-time_plain_tz" on _hyper_2_2_chunk
< (8 rows)
---
>          ->  Sort
>                Sort Key: (time_bucket('@ 1 min'::interval, hyper_1_tz."time")) DESC
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1_tz
>                            ->  Seq Scan on _hyper_2_2_chunk
> (9 rows)
306,307c316,317
<                                                        QUERY PLAN                                                        
< -------------------------------------------------------------------------------------------------------------------------
---
>                                                     QUERY PLAN                                                     
> -------------------------------------------------------------------------------------------------------------------
311,316c321,327
<          ->  Result
<                ->  Merge Append
<                      Sort Key: (time_bucket('@ 1 min'::interval, (hyper_1_tz."time")::timestamp without time zone)) DESC
<                      ->  Index Scan using time_plain_tz on hyper_1_tz
<                      ->  Index Scan using "2-time_plain_tz" on _hyper_2_2_chunk
< (8 rows)
---
>          ->  Sort
>                Sort Key: (time_bucket('@ 1 min'::interval, (hyper_1_tz."time")::timestamp without time zone)) DESC
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1_tz
>                            ->  Seq Scan on _hyper_2_2_chunk
> (9 rows)
328,329c339,340
<                                    QUERY PLAN                                    
< ---------------------------------------------------------------------------------
---
>                            QUERY PLAN                            
> -----------------------------------------------------------------
333,340c344,352
<          ->  Result
<                ->  Merge Append
<                      Sort Key: (((hyper_1_int."time" / 10) * 10)) DESC
<                      ->  Index Scan using time_plain_int on hyper_1_int
<                      ->  Index Scan using "3-time_plain_int" on _hyper_3_3_chunk
<                      ->  Index Scan using "4-time_plain_int" on _hyper_3_4_chunk
<                      ->  Index Scan using "5-time_plain_int" on _hyper_3_5_chunk
< (10 rows)
---
>          ->  Sort
>                Sort Key: (((hyper_1_int."time" / 10) * 10)) DESC
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1_int
>                            ->  Seq Scan on _hyper_3_3_chunk
>                            ->  Seq Scan on _hyper_3_4_chunk
>                            ->  Seq Scan on _hyper_3_5_chunk
> (11 rows)
352,353c364,365
<                                     QUERY PLAN                                     
< -----------------------------------------------------------------------------------
---
>                                  QUERY PLAN                                  
> -----------------------------------------------------------------------------
357,364c369,377
<          ->  Result
<                ->  Merge Append
<                      Sort Key: (((((hyper_1_int."time" - 2) / 10) * 10) + 2)) DESC
<                      ->  Index Scan using time_plain_int on hyper_1_int
<                      ->  Index Scan using "3-time_plain_int" on _hyper_3_3_chunk
<                      ->  Index Scan using "4-time_plain_int" on _hyper_3_4_chunk
<                      ->  Index Scan using "5-time_plain_int" on _hyper_3_5_chunk
< (10 rows)
---
>          ->  Sort
>                Sort Key: (((((hyper_1_int."time" - 2) / 10) * 10) + 2)) DESC
>                ->  Result
>                      ->  Append
>                            ->  Seq Scan on hyper_1_int
>                            ->  Seq Scan on _hyper_3_3_chunk
>                            ->  Seq Scan on _hyper_3_4_chunk
>                            ->  Seq Scan on _hyper_3_5_chunk
> (11 rows)
417,418c430,431
<                                           QUERY PLAN                                           
< -----------------------------------------------------------------------------------------------
---
>                                                 QUERY PLAN                                                 
> -----------------------------------------------------------------------------------------------------------
420,424c433,441
<    ->  GroupAggregate
<          Group Key: date_trunc('minute'::text, "time")
<          ->  Index Scan using time_plain_plain_table on plain_table
<                Index Cond: ("time" < 'Wed Dec 31 16:15:00 1969 PST'::timestamp with time zone)
< (5 rows)
---
>    ->  Sort
>          Sort Key: (date_trunc('minute'::text, "time")) DESC
>          ->  HashAggregate
>                Group Key: date_trunc('minute'::text, "time")
>                ->  Bitmap Heap Scan on plain_table
>                      Recheck Cond: ("time" < 'Wed Dec 31 16:15:00 1969 PST'::timestamp with time zone)
>                      ->  Bitmap Index Scan on time_plain_plain_table
>                            Index Cond: ("time" < 'Wed Dec 31 16:15:00 1969 PST'::timestamp with time zone)
> (9 rows)
